cdap-spark-core/src/main/scala/co/cask/cdap/app/runtime/spark/DatasetCompute.scala

```diff
@@ -35,5 +35,6 @@ private[spark] trait DatasetCompute {
     * @tparam T type of the result
     * @return result of the computation by the function
     */
-  def apply[T: ClassTag](datasetName: String, arguments: Map[String, String], computeFunc: Dataset => T): T
+  def apply[T: ClassTag](datasetNamespace: String, datasetName: String, arguments: Map[String, String],
```

Comment:
so this doesn't need a method that does not accept a ``datasetNamespace`` and uses the program's namespace?